### ISSUE TITLE
Disable fastlane run when not executed via `bundle exec`

### ISSUE THREAD

**Issue by DominikBucher12** [fastlane/fastlane#13857]
#### Motivation Behind Feature
When you are using Fastlane with bundler you can get easily distracted by not typing `bundle exec fastlane` and do simply `fastlane` This is cool but I want to run the fastlane just via `bundle exec` and not via global fastlane, this could lead to some problems with the installation

#### Feature Description
#9435 Describes this totally awesome 

#### Alternatives or Workarounds

Use this `before_all` lane 

```
before_all do |lane|
         unless ENV.key?("BUNDLE_BIN_PATH")
             UI.user_error!("Please run using bundle exec fastlane.")
         end
end
```
or make alias to `fastlane` 🗡  `alias fastlane="bundle exec fastlane"` which totally sucks

I can work on this feature as long as it won't get auto-closed again :D Thanks for any response and not closing this :D

**Comment by DominikBucher12** [comment]
Actually... This is not solution to the proposed problem... Looks like I got confused... The action is just nicer way to the workaround I provided... The thing is I would like to have `Fastlane` To automatically abort before prompting for any lane... It still runs while using just `fastlane` command and after choosing lane it obviously throws error because it reaches `before_all` lane...

**Comment by joshdholtz** [comment]
@DominikBucher12 Using this action is the best we can do for you. This seems like it is a pretty specific use case. Not everybody uses `bundle exec` (a lot of users will have _fastlane_ installed globally in there system). This is something that we would _technically_ be able to put into fastlane core but not something that could be enabled by default so it would have to be a flag/variable that gets set. I'm also guessing here that most users would be totally fine with adding an the optional `ensure_bundle_exec` action here as suggested in #9435 and implemented in #13866.

### TL;DR
Technically possible to add to fastlane core but not core enough of a feature to put there so best implementation is through an action.

### Sidenote
I totally use `alias fl="bundle exec fastlane"` because it adds `bundle exec` and its shorter to type.

**Comment by DominikBucher12** [comment]
Well... I don't find this `pretty specific use case` Then you could say it's not worth it to point out to user that he is using Fastlane globally and not via `bundle exec` while he has Gemfile upon Fastlane startup.

**Comment by joshdholtz** [comment]
@DominikBucher12 Not all users have a `Gemfile` 😊 If installed via `homebrew` or with `gem install fastlane`, a user could use _fastlane_ completely fine without `bundle exec`. Many users don't even have a `Fastfile` either. These users will run the tools/actions directly with `fastlane <tool/action>` and so it would be overkill to point out to the user to create a `Gemfile` and use `bundle exec`. That being said...

Do you not have a `Fastfile` where you could make use of the `ensure_bundle_exec`? Or does getting to that point take too long due to some other work _fastlane_ does before it gets to executing of the `Fastfile`? I'm just trying to understand why the approaches mentioned above won't work for your setup 🙃

**Comment by DominikBucher12** [comment]
I totally got your point, thanks for your time :) You are right about this, I couldn't just get this picture. closing this since this feature is almost useless for most of the users.
